src/lib.rs

```diff
@@ -216,6 +224,15 @@ cfg_if! {
         #[path = "solaris_illumos.rs"] mod imp;
     } else if #[cfg(target_os = "wasi")] {
         #[path = "wasi.rs"] mod imp;
+    } else if #[cfg(any(
+        // target_vendor was stabilized only in Rust 1.33
+        target = "i686-uwp-windows-gnu",
+        target = "x86_64-uwp-windows-gnu",
+        target = "aarch64-uwp-windows-msvc",
+        target = "x86_64-uwp-windows-msvc",
+        target = "i686-uwp-windows-msvc",
```

Comment:
Looks like this will not work, see rust-lang/rust#63217. :/ So our choices are:
- Do not support UWP
- Bump MSRV to Rust 1.33
- Add `uwp` or `rust_1_33` feature.

Reply:
I think we should bump the MSRV to 1.33 (and release a new minor version), Rust 1.38 is coming out in a few days (Aug 12 I think), so bumping to 1.33 then seems reasonable.

Reply:
Personally until something like rust-lang/rfcs#2495 lands I prefer a conservative position stating that MSRV bump should be considered a breaking change.

I think we also could move UWP check to the Windows branch, so MSRV bump will affect only Windows users.

@dhardy 
What dou you think?

Reply:
After looking at #75 and rust-lang/rfcs#2495, I think you're right. Incrementing the minimum supported version _should be considered a breaking change_. However, we can support UWP without bumping the MSRV, adding a feature, or dropping support for XP/Vista. We can just check in the build script.

[`libc` does this](https://github.com/rust-lang/libc/blob/master/build.rs) to maintain a comically low MSRV (Rust 1.13). They just check for the version in `build.rs` and then set `cfg`s appropriately.

So we would do the following:
```rust
    ...
    if target.contains("uwp") {
        println!("cargo:rustc-cfg=getrandom_uwp");
        // for BCryptGenRandom
        println!("cargo:rustc-link-lib=bcrypt");
    ...
```

```rust
// src/lib.rs
    ...
    } else if #[cfg(all(windows, getrandom_uwp))] {
        #[path = "windows_uwp.rs"] mod imp;
    } else if #[cfg(windows)] {
        #[path = "windows.rs"] mod imp;
    }
    ...
```

This prevents exposing any of the details in `getrandom`'s public interface.

Reply:
Good workaround!

Reply:
Using a build.rs file seems appropriate in this case.